### ISSUE TITLE
Magento\Search\Helper\getSuggestUrl() not used in search template

### ISSUE THREAD

**Issue by schmengler** [magento/magento2#6802]
### Preconditions

1. Magento 2.1.1
### Steps to reproduce

1.  Create a plugin for Magento\Search\Helper\getSuggestUrl() to add custom autosuggest feature
2. Type something in the search box
### Expected result

1. the custom URL should be used
### Actual result

1. the default "search/ajax/suggest" URL is used

The problem is that `getSuggestUrl()` is never used anywhere in the core code. Instead, `magento/module-search/view/frontend/templates/form.mini.phtml` contains the hard coded route:

```
echo $block->getUrl('search/ajax/suggest', ['_secure' => $block->getRequest()->isSecure()]);
```

This makes it hard to implement a useful autosuggest feature, especially since it might very well be a URL that's not going to a Magento controller.

**Comment by magento-engcom-team** [comment]
@schmengler, thank you for your report.
We've created internal ticket(s) MAGETWO-81592 to track progress on the issue.

**Comment by okorshenko** [comment]
The issue has been fixed in 2.2-develop branch. Will be available with 2.2.2 release

**Comment by magento-team** [comment]
Hi @schmengler. Thank you for your report. 
The issue has been fixed in magento-engcom/magento2ce#1287 by @magento-engcom-team in 2.3-develop branch
Related commit(s):


The fix will be available with the upcoming patch release.
